src/QMCDrivers/QMCFixedSampleLinearOptimize.cpp

```diff
@@ -528,8 +531,12 @@ bool QMCFixedSampleLinearOptimize::put(xmlNodePtr q)
 #endif
       optTarget = new QMCCostFunction(W, Psi, H, myComm);
     optTarget->setStream(&app_log());
+    if (ReportToH5=="yes")
+       optTarget->reportH5=true;
     success = optTarget->put(q);
   }
+  if (ReportToH5=="yes")
```

Comment:
Could you explain why these two lines needs to be repeated?

Reply:
The writing of the CI from H5 is activated by the tag hdf5=yes 
 
it can only be read in this function bool QMCFixedSampleLinearOptimize::put(xmlNodePtr q)
but it is needed to write from optTarget.  

optTarget needs to be created before we can  pass the value of ReportH5. OptTarget is created only once but if you have multilple optimization blocks you can decide to save only a few blocks to H5. We therefore need to read again the value hdf5 and pass it to optTarget. hence one definition in the if optTarget==0.
Now in the optTarget==0 there is a call to 
    success = optTarget->put(q);
this function needs to know if ReportH5 is yes or no before being called. This is why i cannot use just the one call outside the if statement.